src/db/migrations/22-add-zora-creator.ts

```diff
@@ -0,0 +1,35 @@
+import { Knex } from 'knex';
+
+import { MetaFactory, MetaFactoryTypeName } from '../../types/metaFactory';
+import { NFTStandard } from '../../types/nft';
+import { Table } from '../db';
+import { addMetaFactory, removeMetaFactory } from '../migration-helpers';
+
+const ZORA_CREATOR_FACTORY: MetaFactory = 
+  {
+    address: '0xf74b146ce44cc162b601dec3be331784db111dc1',
+    platformId: 'zora',
+    startingBlock: '14758779',
+    contractType: MetaFactoryTypeName.zoraDropCreator,
+    gap: '500000',
+    standard: NFTStandard.ERC721,
+    autoApprove: false
+  }
+
+export const up = async (knex: Knex) => {
+  await addMetaFactory(knex, ZORA_CREATOR_FACTORY);
+
+  const hasColumn = await knex.schema.hasColumn(Table.erc721Transfers, 'transactionHash')
+  if (!hasColumn){
+    await knex.schema.alterTable(Table.erc721Transfers, table => {
+      table.string('transactionHash');
+      table.dropPrimary(`${Table.erc721Transfers}_pkey`)
+      table.primary(['id', 'from', 'to', 'contractAddress', 'tokenId', 'transactionHash'])
```

Comment:
this change doesn't work due to this issue:
 - column "transactionHash" of relation "raw_erc721_transfers" contains null values

if we want to use transactionHash in the primary key then i guess we should do a full rebuild anyway, so this optional migration is probably unnecessary

Reply:
alternatively we could leave it out the primary key, then fill it in, then migrate it into a primary key after

Reply:
maybe I could also make transactionHash nullable?

Reply:
hmm i would expect this primary key to break in the following scenario anyway tho:
 - if someone sends an nft from person a -> b -> a again all in the same transaction, then all 6 of those fields will be the same even though there were 2 different transfer events

and so i would expect a crash again in this scenario

this is why i used the event index as part of the ID to try be super robust/futureproof, so it is weird that we're getting the collision issue. maybe there's a bad assumption i'm making or some bug somewhere. 

did you find out where the pk conflict came from yet?

Reply:
even in that case wouldn't the log index be different? I'm not sure in exactly which case this was happening but I'll try to find an example

Reply:
yeah the log index will be different, so that's why the log index seems important to use in the pk